extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ClientProxyUnwrapper.java

```diff
@@ -7,9 +7,6 @@
 public class ClientProxyUnwrapper implements Function<Object, Object> {
     @Override
     public Object apply(Object o) {
-        if (o instanceof ClientProxy) {
-            return ((ClientProxy) o).arc_contextualInstance();
-        }
-        return o;
+        return ClientProxy.unwrap(o);
```

Comment:
Shouldn't we deprecate this method (and/or class) and point to the new static method?
And when we do that, replace any usage in Quarkus itself.

Reply:
That's a good point. I only found `ProxyUnwrapperBuildItem` which is not used in quarkus core but is considered a part of the public API.